static/js/timerender.ts

```diff
@@ -306,56 +306,22 @@ export function stringify_time(time: number | Date): string {
 
 // this is for rendering absolute time based off the preferences for twenty-four
 // hour time in the format of "%mmm %d, %h:%m %p".
-export const absolute_time = (function () {
-    const MONTHS = [
-        "Jan",
-        "Feb",
-        "Mar",
-        "Apr",
-        "May",
-        "Jun",
-        "Jul",
-        "Aug",
-        "Sep",
-        "Oct",
-        "Nov",
-        "Dec",
-    ];
-
-    const fmt_time = function (date: Date, H_24: boolean): string {
-        const payload = {
-            hours: date.getHours(),
-            minutes: date.getMinutes(),
-            is_pm: false,
-        };
-
-        if (payload.hours > 12 && !H_24) {
-            payload.hours -= 12;
-            payload.is_pm = true;
-        }
-
-        let str = ("0" + payload.hours).slice(-2) + ":" + ("0" + payload.minutes).slice(-2);
-
-        if (!H_24) {
-            str += payload.is_pm ? " PM" : " AM";
-        }
-
-        return str;
-    };
-
-    return function (timestamp: number, today = new Date()): string {
-        const date = new Date(timestamp);
-        const is_older_year = today.getFullYear() - date.getFullYear() > 0;
-        const H_24 = user_settings.twenty_four_hour_time;
-        let str = MONTHS[date.getMonth()] + " " + date.getDate();
-        // include year if message date is from a previous year
-        if (is_older_year) {
-            str += ", " + date.getFullYear();
-        }
-        str += " " + fmt_time(date, H_24);
-        return str;
-    };
-})();
+export function absolute_time(timestamp: number, today = new Date()): string {
+    const date = new Date(timestamp);
+    const is_older_year = today.getFullYear() - date.getFullYear() > 0;
+    const H_24 = user_settings.twenty_four_hour_time;
+
+    return format(
+        date,
+        is_older_year
+            ? H_24
+                ? "MMM d, yyyy HH:mm"
+                : "MMM d, yyyy hh:mm a"
+            : H_24
+            ? "MMM d HH:mm"
+            : "MMM d hh:mm a",
+    );
+}
```

Comment:
Are there any cases where we expect this to produce different output than the previous implementation?  I'm not sure if we have tests for this code, but it seems like we should at least discuss/document changes in display format.

Reply:
No, this `absolute_time` implementation should work identically. In the future we should make it locale-aware, but this one isn’t because date-fns defaults to `en-US` and the format strings aren’t sufficiently localized.

The `blueslip` logging format is slightly different (`2021-09-22 22:56:13.708 UTC` → `2021-09-22T22:56:13.708Z`), but for logging I assume that’s not important.

Reply:
Yeah, I don't care about the logging format changing as long as its readable.  I do think that format is substantially less human-readable with the T rather than a space, but the timestamp section of blueslip log lines are not consulted often, so it's not important, and we can always tweak further if we start looking at them often.